Sources/AblyAssetTrackingInternal/WorkerQueue/LegacyWorker.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+/// A worker that runs tasks that the Publisher and Subscriber used to run asynchronously on their
+/// own dispatch queues. This worker allows us to tie up legacy work and new asynchronous
+/// work, for simplicity and consistency during the transition.
+open class LegacyWorker<PropertiesType, WorkerSpecificationType> : Worker
+{
+    let work: () -> Void
+
+    public init (work: @escaping () -> Void) {
+        self.work = work
+    }
+
+    public func doWork(properties: PropertiesType, doAsyncWork: (@escaping () throws -> Void) -> Void, postWork: @escaping (WorkerSpecificationType) -> Void) throws -> PropertiesType {
```

Comment:
It'd be good to see an example of the worker’s ability to log something – could we pass a log handler here from the factory and log some simple message like "Performing work function"?

Reply:
Agreed, done in e94e433 :)